lib/game.rb

```diff
@@ -0,0 +1,102 @@
+class Game < Board
+  attr_accessor :board, :player_1, :player_2
+WIN_COMBINATIONS = [
+[0,1,2],
+[3,4,5],
+[6,7,8],
+[0,4,8],
+[2,4,6],
+[0,3,6],
+[1,4,7],
+[2,5,8]
+]
+
+  def initialize (player_1 = nil, player_2 = nil, board = nil)
+    @player_1 = player_1 || @player_1 = Players::Human.new("X")
+    @player_2 = player_2 || @player_2 = Players::Human.new("O")
+    @board = board || @board = Board.new
+  end
+  
+  def current_player
+    if @board.turn_count.even?
+      @player_1
+    else 
+      @player_2
+   end 
+  end 
+  
+  def won?
+    WIN_COMBINATIONS.detect do |win| 
+      @board.cells[win[0]] == @board.cells[win[1]] && @board.cells[win[0]] == @board.cells[win[2]] && (@board.cells[win[0]] == "X" || @board.cells[win[0]] == "O")
+    end
+  end
+  
+  def draw?
+    !won? && @board.full?
+  end
+  
+  def over?
+     won? || draw?
+  end
+  
+  def winner 
+    if won? == nil
+      nil
+    else
+      @board.cells[won?[0]]
+    end
+  end
+  
+  def turn 
+    num = current_player.move(board)
+      if @board.valid_move?(num)
+        @board.update(num, current_player)
+        @board.display
+      else 
+        turn
+      end
+  end
+
+  def play
+    until over?
+    turn
+    end
+      if won?
+        puts "Congratulations #{winner}!"
+      elsif draw?
+        puts "Cat's Game!"
+      end
+  end
+    
+  def start 
```

Comment:
Great bin file, short length is ideal. However, we would prefer to see the CLI logic separated from the game logic in its own class (e.g. TicTacToeCLI). Splitting the interface from the implementation of the game allows for multiple interfaces (e.g. a web app) without requiring changes to the game code.